File: scripts/generateQRCodes.js
require('dotenv').config();
const fs = require('fs');
const path = require('path');
const QRCode = require('qrcode');
const { connect } = require('../db');
const ContainerService = require('../services/containerService');

async function run() {
  await connect();
  const containers = await ContainerService.listContainers();
  const dir = path.join(__dirname, '..', 'qrcodes');
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir);
  }
  for (const c of containers) {
    const file = path.join(dir, `${c.uuid}.png`);
    await QRCode.toFile(file, c.uuid);
    console.log('Generated QR for', c.uuid);
  }
  process.exit();
}

run().catch(err => {
  console.error(err);
  process.exit(1);
});

File: scripts/generate_qrcodes.py
import os
import pathlib
import qrcode
from services import container_service


def run():
    containers = container_service.list_containers()
    out_dir = pathlib.Path("qrcodes")
    out_dir.mkdir(exist_ok=True)
    for c in containers:
        file_path = out_dir / f"{c['uuid']}.png"
        qrcode.make(c["uuid"]).save(file_path)
        print("Generated QR for", c["uuid"])


if __name__ == "__main__":
    run()
